src/main/java/com/microsoft/graph/serializer/CollectionPageSerializer.java

```diff
@@ -0,0 +1,138 @@
+import com.google.gson.JsonParser;
+import com.google.common.reflect.TypeToken;
+import com.microsoft.graph.http.BaseCollectionPage;
+import com.microsoft.graph.http.IRequestBuilder;
+import com.microsoft.graph.logger.ILogger;
+
+// those imports are useless but build will fail if code-gen conventions change, keep it
+import com.microsoft.graph.models.extensions.Attachment;
+import com.microsoft.graph.requests.extensions.AttachmentCollectionPage;
+import com.microsoft.graph.requests.extensions.AttachmentCollectionResponse;
+import com.microsoft.graph.requests.extensions.IAttachmentCollectionRequestBuilder;
+
+public class CollectionPageSerializer {
+
+	private static DefaultSerializer serializer;
+	/** length of the word "page" */
+	private final static Integer pageLength = 4;
+	/** length of the word "collection" */
+	private final static Integer collectionLength = 10;
+	/** length of the work "response" */
+	private final static Integer responseLength = 8;
+	/** the extensions segment in the package name of target classes */
+	private final static String extensionsPath = "extensions.";
+
+	/**
+	 * Not available for instantiation
+	 */
+	private CollectionPageSerializer() {
+	}
+
+	/**
+     * Serializes an CollectionPage
+     * 
+     * @param src the CollectionPage variable for serialization
+     * @param logger the logger
+     * @return       JsonElement of CollectionPage
+     */
+	public static <T1, T2 extends IRequestBuilder> JsonElement serialize(final BaseCollectionPage<T1, T2> src, final ILogger logger) {
+		if(src == null) {
+			return null;
+		}
+		JsonArray jsonArray = new JsonArray();
+		List<T1> items = src.getCurrentPage();
+		serializer = new DefaultSerializer(logger);
+		for(T1 item : items) {
+			final String json = serializer.serializeObject(item);
+			final JsonElement element = JsonParser.parseString(json);
+			if(element != null && element.isJsonObject()) {
+				final JsonObject jsonObject = element.getAsJsonObject();
+				jsonArray.add(jsonObject);
+			}
+		}
+		return jsonArray;
+	}
+
+	/**
+     * Deserializes the JsonElement
+     *
+     * @param json the source CollectionPage's Json
+   	 * @param typeOfT The type of the CollectionPage to deserialize to
+     * @param logger the logger
+     * @throws JsonParseException the parse exception
+     * @return    the deserialized CollectionPage
+     */
+	@SuppressWarnings("unchecked")
+	public static <T1, T2 extends IRequestBuilder> BaseCollectionPage<T1, T2> deserialize(final JsonElement json, Type typeOfT, final ILogger logger) throws JsonParseException {
+		if (json == null) {
+			return null;
+		}
+		serializer = new DefaultSerializer(logger);
+		final JsonObject[] sourceArray = serializer.deserializeObject(json.toString(), JsonObject[].class);
+		final ArrayList<T1> list = new ArrayList<T1>(sourceArray.length);
+		final String collectionPageClassCanonicalName = typeOfT.getTypeName();
+		final String entityClassCanonicalName = collectionPageClassCanonicalName
```

Comment:
I don't understand this string manipulation. Can you add a comment to explain why this is done? This occurs in a few places so this comment applies to those places as well.

Reply:
added comments